services/steam/steam.service.js

```diff
@@ -0,0 +1,395 @@
+}
+
+class SteamFileSize extends SteamFileService {
+  static render({ fileSize }) {
+    return { message: prettyBytes(fileSize), color: 'green' }
+  }
+
+  async onRequest({ response }) {
+    return this.constructor.render({ fileSize: response.file_size })
+  }
+
+  static get category() {
+    return 'size'
+  }
+
+  static get defaultBadgeData() {
+    return { label: 'size' }
+  }
+
+  static get url() {
+    return {
+      base: 'steam/size',
+      format: '(.+)',
+      capture: ['fileId'],
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'Steam File Size',
+        exampleUrl: '100',
+        urlPattern: ':file_id',
+        staticExample: this.render({ fileSize: 20000 }),
+        keywords: ['steam'],
+      },
+    ]
+  }
+}
+
+class SteamReleaseDate extends SteamFileService {
+  static render({ releaseDate }) {
+    return { message: formatDate(releaseDate), color: ageColor(releaseDate) }
+  }
+
+  async onRequest({ response }) {
+    const releaseDate = new Date(0).setUTCSeconds(response.time_created)
+    return this.constructor.render({ releaseDate: releaseDate })
+  }
+
+  static get defaultBadgeData() {
+    return { label: 'release date' }
+  }
+
+  static get url() {
+    return {
+      base: 'steam/release-date',
+      format: '(.+)',
+      capture: ['fileId'],
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'Steam Release Date',
+        exampleUrl: '100',
+        urlPattern: ':file_id',
+        staticExample: this.render({
+          releaseDate: new Date(0).setUTCSeconds(1538288239),
+        }),
+        keywords: ['steam'],
+      },
+    ]
+  }
+}
+
+class SteamSubscriptions extends SteamFileService {
+  static render({ subscriptions }) {
+    return { message: metric(subscriptions), color: 'lime' }
+  }
+
+  async onRequest({ response }) {
+    return this.constructor.render({ subscriptions: response.subscriptions })
+  }
+
+  static get defaultBadgeData() {
+    return { label: 'subscriptions' }
+  }
+
+  static get url() {
+    return {
+      base: 'steam/subscriptions',
+      format: '(.+)',
+      capture: ['fileId'],
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'Steam Subscriptions',
+        exampleUrl: '100',
+        urlPattern: ':file_id',
+        staticExample: this.render({ subscriptions: 20124 }),
+        keywords: ['steam'],
+      },
+    ]
+  }
+}
+
+class SteamFavorites extends SteamFileService {
+  static render({ favorites }) {
+    return { message: metric(favorites), color: 'lime' }
+  }
+
+  async onRequest({ response }) {
+    return this.constructor.render({ favorited: response.favorited })
+  }
+
+  static get defaultBadgeData() {
+    return { label: 'favorites' }
+  }
+
+  static get url() {
+    return {
+      base: 'steam/favorites',
+      format: '(.+)',
+      capture: ['fileId'],
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'Steam Favourites',
```

Comment:
I note you've gone with the UK spelling "favourite" in the description but the American "favorite" on the badge and everywhere else. Can we go with the American style consistently.